exercises/guillaume_communie/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,40 @@
+#include "ShapeSorter.h"
```

Comment:
A general comment for this class. You are doing std::cout in each method, which is the same line more or less repeated. It would be better to have a method that receives a vector and prints all of it, and then the filter methods, which will filter the input vector based on the corresponding criteria and then pass over the printer method. You could also use std::find_if or similar for the filters.

Reply:
* I added a method to print a vector of shapes
* for the second point, I do not really know how to do it. For the first method (`printMatchType`) I used `std::copy_if` to create a new vector and print it with the previously mentioned method. Otherwise, `std::find_if` would have to be called several times to print all the solution and not the first one (?)

Reply:
Indeed find finds only the first one.